network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/CachedNetworkStoreClient.java

```diff
@@ -29,105 +29,245 @@ public class CachedNetworkStoreClient extends AbstractForwardingNetworkStoreClie
-        delegate::getShuntCompensators));
-
-    private final NetworkCollectionIndex<CollectionCache<VscConverterStationAttributes>> vscConverterStationCache = new NetworkCollectionIndex<>(() -> new CollectionCache<>(
-        delegate::getVscConverterStation,
-        delegate::getVoltageLevelVscConverterStations,
-        delegate::getVscConverterStations));
-
-    private final NetworkCollectionIndex<CollectionCache<LccConverterStationAttributes>> lccConverterStationCache = new NetworkCollectionIndex<>(() -> new CollectionCache<>(
-        delegate::getLccConverterStation,
-        delegate::getVoltageLevelLccConverterStations,
-        delegate::getLccConverterStations));
-
-    private final NetworkCollectionIndex<CollectionCache<StaticVarCompensatorAttributes>> staticVarCompensatorCache = new NetworkCollectionIndex<>(() -> new CollectionCache<>(
-        delegate::getStaticVarCompensator,
-        delegate::getVoltageLevelStaticVarCompensators,
-        delegate::getStaticVarCompensators));
-
-    private final NetworkCollectionIndex<CollectionCache<HvdcLineAttributes>> hvdcLinesCache = new NetworkCollectionIndex<>(() -> new CollectionCache<>(
-        delegate::getHvdcLine,
-        null,
-        delegate::getHvdcLines));
-
-    private final NetworkCollectionIndex<CollectionCache<DanglingLineAttributes>> danglingLinesCache = new NetworkCollectionIndex<>(() -> new CollectionCache<>(
-        delegate::getDanglingLine,
-        delegate::getVoltageLevelDanglingLines,
-        delegate::getDanglingLines));
-
-    private final NetworkCollectionIndex<CollectionCache<ConfiguredBusAttributes>> configuredBusesCache = new NetworkCollectionIndex<>(() -> new CollectionCache<>(
-        delegate::getConfiguredBus,
-        delegate::getVoltageLevelConfiguredBuses,
-        delegate::getConfiguredBuses));
-
-    private final NetworkCollectionIndex<CollectionCache<TieLineAttributes>> tieLinesCache = new NetworkCollectionIndex<>(() -> new CollectionCache<>(
-            delegate::getTieLine,
-            null,
-            delegate::getTieLines));
-
-    private final NetworkCollectionIndex<CollectionCache<GroundAttributes>> groundsCache = new NetworkCollectionIndex<>(() -> new CollectionCache<>(
-            delegate::getGround,
-            delegate::getVoltageLevelGrounds,
-            delegate::getGrounds));
+    private final NetworkCollectionIndex<CollectionCache<NetworkAttributes>> networksCache =
+            new NetworkCollectionIndex<>(() -> new CollectionCache<>(
+                    (networkUuid, variantNum, id) -> delegate.getNetwork(networkUuid, variantNum),
+                    null,
+                    (networkUuid, variantNum) -> delegate.getNetwork(networkUuid, variantNum).stream().collect(Collectors.toList()),
+                    delegate::getExtensionAttributes,
+                    delegate::getAllExtensionsAttributesByResourceTypeAndExtensionName,
+                    delegate::getAllExtensionsAttributesByIdentifiableId,
```

Comment:
should not copypaste the same method everywhere

Reply:
Simplified this in https://github.com/powsybl/powsybl-network-store/commit/1d2c57e75f0193c6d358372294460f2872ac1d4d
I passed the delegate instead of all the methods.